Feature/so nak UI

choo121600 opened the pull request:
## Motivation ⍰
### PlayerNameInputView
- [X] PlayerNameInputView 에 입력 값 감지 테두리 UI 변경 (회색 -> 밸런스 블루)
- [X] 폰트 변경
- [X] TextField Label 우측 정렬
- [X] safeArea 적용
- [X] Player의 수가 많은 경우 ScrollView처리 -> next버튼 보이게 처리

### SelectQuestionThemeView
- [X] 폰트 배경색
- [X] 셀 중복선택 불가 처리
- [X] 셀 선택시 배경색 변경(흰 -> 하늘색)
- [X] 스크롤 뷰 처리
- [X] Next버튼을 누르면 SelectTypeView로 selectedTheme값 이동 

<br>

## Key Changes 🔑

- 

<br>

## To Reviewers 🙏🏻
### PlayerNameInputView
- [x] PlayerNameInputView 에 입력 값 감지 UI 변경
- [x] 폰트 변경
- [ ] TextField Label 우측 정렬
- [x] safeArea 적용
- [x] Player의 수가 많은 경우 ScrollView처리 -> next버튼 보이게 처리

### SelectQuestionThemeView
- [ ] 폰트 배경색
- [x] 셀 중복선택 불가 처리
- [x] 셀 선택시 배경색 변경(흰 -> 하늘색)
- [x] 스크롤 뷰 처리
- [x] Next버튼을 누르면 SelectTypeView로 selectedTheme값 이동 

<br>

## Linked Issue 🔗

- [x] #8

choo121600 committed:
Feat: Geometry Width Resizing in BeforePlayRuleDescriptionView

choo121600 committed:
Feat: Geometry Width Resizing in BeforePlayRuleDescriptionView

choo121600 committed:
Feat: PlayerNameInputView & SelectQuestionThemeView 작업

Commit:
폰트 테두리 추가, 최후변론 view 추가

choo121600 committed:
Fix: WhoIsLoserView에서 SelectQuestionThemeView 연결

choo121600 committed:
Feat: SelectQuestionThemeView UI및 SelectTypeView로 selectedTheme값 전달

Minny27 commented:
1. 룰 설명 뷰에서 **next 버튼 간격**을 Spacer()로 주고 있는데 **디자인 가이드랑 상이**한 것 같습니다.
<img src="https://user-images.githubusercontent.com/68800789/229777591-bc258df9-9ce5-439d-8e1b-35804ac240ac.png" width=25%>

2. 플레이어 이름 입력 뷰에서 인원 수에 따라 next의 위치를 동적으로 가져가자고 이야기 했었던 기억이 있는데 현재는 **버튼의 위치는 고정**되어 있습니다.
<img src="https://user-images.githubusercontent.com/68800789/229779043-a0a2efd9-9ae8-4f36-94b8-6fbc4bd92b58.png" width=25%>

Minny27 commented:
추가적으로 질문 테마 선택하는 화면에서 특정 셀을 클릭했을 때의 **이벤트 영역이 텍스트로 잡혀있는 것** 같습니다.

choo121600 committed:
Fix: 버튼 위치

choo121600 committed:
Fix: theme Button

choo121600 committed:
Fix: 인원 수 입력값이 0이하 처리

choo121600 commented:
### 추가 수정 사항
- [x] 테마 버튼 텍스트 이벤트에서 버튼 이벤트로 수정
- [ ] PlayerNumberInputView 사람 수에 맞춰서 Next버튼 유동적으로 수정
- [x] numberOfPeople이 0 이하인 상태에서 next버튼 눌리면 알림창이 뜬다

Minny27 committed:
Update: 플레이어 이름 적는 화면에서 next 버튼 동적 높이 설정
- 뷰 간격을 가이드에 맞게 Space().frame(height: )으로 구현
- Scroll view의 불필요한 bottom, top padding 삭제
- 플레이어 명수에 따른 scroll view 최대 높이를 스크린의 3 / 2로 설정
- scroll view가 최대 높이 이상이 되면 next버튼을 safe area로 고정 시키기

Minny27 committed:
Update: scroll view bottom 패딩 값 삭제

# 제목의 길이는 최대 40글자까지 한글로 간단 명료하게 작성
# 제목을 작성하고 반드시 빈 줄 한 줄을 만들어야 함 # 제목에 .(마침표) 금지
# <label> 리스트
# Feat: 새로운 기능
# Fix: 버그 수정
# Add: Swift 파일 추가
# Update: Swift 파일 (삭제, 이동, 이름 변경)
# Style: 코드 스타일 혹은 포맷 등에 관한 커밋
# Docs: 문서 (md 파일 추가, 수정, 삭제)
# Refactor: 코드 가독성과 유지 보수의 편의에 관한 커밋
# Test: 테스트
# <description>
# 내용의 길이는 한 줄당 60글자 내외에서 줄 바꿈. 한글로 간단 명료하게 작성
# 어떻게 보다는 무엇을, 왜 변경했는지를 작성할 것 (필수)
# <issue-number>
# 연관된 이슈 첨부, 여러 개 추가 가능

Minny27 commented:
플레이어 이름 입력 화면에서 next버튼 동적 위치 구현했습니다.
미리보기는 다음과 같고 수정 사항은 커밋 메시지에 작성했습니다.
<img src="https://user-images.githubusercontent.com/68800789/230056825-00bd31b4-bc2f-43a1-aeb4-51ffe8c66e17.png" width=25%><img src="https://user-images.githubusercontent.com/68800789/230056450-76c176a1-71ab-4dc5-a171-5016a3486202.png" width=25%> <img src="https://user-images.githubusercontent.com/68800789/230056456-5d688b3d-edca-44fc-993a-ccf287a07b3b.png" width=25%>

Minny27 commented on Balance-Catch-iOS/Balance-Catch-iOS/View/PlayerNameInputView.swift:
```diff
@@ -8,55 +8,93 @@
 }
 
-
+struct ViewHeightKey: PreferenceKey {
```
PreferenceKey로 부모 뷰가 자식 뷰의 동적으로 변경된 높이를 get할 수 있군요. 좋네요

choo121600 committed:
Merge branch 'develop' into Feature/SoNakUI
